benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java

```diff
@@ -207,12 +209,18 @@ public final void runForSync(Function<SELF, Function<Long, Object>> func) {
 
       Function<Long, Object> unitOfWork = func.apply(self);
 
-      Flux<Long> fromStream = Flux.fromStream(LongStream.range(0, settings.numOfIterations()).boxed());
+      CountDownLatch latch = new CountDownLatch(1);
 
-      Flux.merge(fromStream
-          .publishOn(scheduler()).map(unitOfWork))
-          .take(settings.executionTaskDuration())
-          .blockLast();
+      Flux.fromStream(LongStream.range(0, settings.numOfIterations()).boxed())
+          .parallel()
+          .runOn(scheduler())
+          .map(unitOfWork)
+          .doOnTerminate(latch::countDown)
+          .subscribe();
+
+      latch.await(settings.executionTaskDuration().toMillis(), TimeUnit.MILLISECONDS);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
Better just propagate with `Exceptions.propagate(e)` .   Can u put right after this comment an improvement  whihch this parallel  thing is giving to us.  I heard u told it's like 2x perf boost. Ok, can u put the stats right here. Like, here it was, here it became.